### PR TITLE
SRX-M5M4Y3 [kuberpult] [timebox 6h] Setup documentation for new developers

### PR DESCRIPTION
**Pull request by mnishamk-freiheit**

### CODE REVIEW

File: README_DEVELOPERS.md [hunk 1]
```diff
@@ -1,33 +1,144 @@
...
+afterwards run `make release`. this will push the docker image, package the helm chart and create a git tag. the helm chart must be uploaded manually to the github release at the moment.
+afterwards bump the version in the `version` file.
+
+## Notes
 
-## libgit2 vs. ...
+- there is a dev image based on alpine in `docker/build`. you can create a shell using the `./dmake` command.
 
-The first version of this tool was written using go-git v5. Sadly the performance was abysmal. Adding a new manifest took > 20 seconds. Therefore, we switched to libgit2 which is much faster but less ergonomic.
+- the first version of this tool was written using go-git v5. sadly the performance was abysmal. adding a new manifest took > 20 seconds. therefore, we switched to libgit2 which is much faster but less ergonomic.
```
Comment (sven-urbanski-freiheit-com) [inline README_DEVELOPERS.md:144]: please keep it uppercase

File: README_DEVELOPERS.md [hunk 2]
```diff
@@ -1,33 +1,144 @@
...
+## releasing a new version
+
+building with libgit2 is tricky atm. run `./dmake make -c services/cd-service bin/main` once to generate the binary for the cd-service.
+afterwards run `make release`. this will push the docker image, package the helm chart and create a git tag. the helm chart must be uploaded manually to the github release at the moment.
+afterwards bump the version in the `version` file.
+
+## Notes
 
-## libgit2 vs. ...
+- there is a dev image based on alpine in `docker/build`. you can create a shell using the `./dmake` command.
```
[the remark this answers is not in the document]
Comment (mnishamk-freiheit) [reply README_DEVELOPERS.md:142]: This was verbatim from the old readme, will update

File: README_DEVELOPERS.md [hunk 3]
```diff
@@ -1,17 +1,87 @@
 # kuberpult Readme for developers
 
-## Release a new version
+## introduction
 
-Building with libgit2 is tricky atm. Run `./dmake make -C services/cd-service bin/main` once to generate the binary for the cd-service.
-Afterwards run `make release`. This will push the docker image, package the helm chart and create a git tag. The helm chart must be uploaded manually to the github release at the moment.
-Afterwards bump the version in the `version` file.
+kuberpult watches a remote repository, unlike ArgoCD it is not triggered based on push to the repository, It is triggered by rest api instead (or ui which in turn calls the rest api).
```
Comment (hannesg) [inline README_DEVELOPERS.md:5]: kuberpult doesnt watch the repository at all. Instead it uses git as data storage. You also would say that an application watches MySQL or Cassandra.

File: README_DEVELOPERS.md [hunk 4]
```diff
@@ -1,17 +1,87 @@
...
+For a more verbose version, you could go into the service directory and run the tests manually in verbose mode
 
-  Download tar file and follow instructions here: https://github.com/libgit2/libgit2#installation
-  It worked for me to run: (the instructions are slightly different)
+```bash
+cd services/cd-service
+go test ./... -v
+```
+
+# Legacy docs
```
Comment (hannesg) [inline README_DEVELOPERS.md:73]: How is this legacy? We don't intend to get rid of yarn or node. I would call this "Installation outside of docker"

**Review by hannesg — APPROVED** [review]

**Pull request merged**